cluster/v1alpha1/types.go

```diff
@@ -104,3 +104,54 @@ type ManagedClusterSetBindingList struct {
 	// Items is a list of ManagedClusterSetBinding.
 	Items []ManagedClusterSetBinding `json:"items"`
 }
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:resource:scope="Cluster"
+
+// ClusterClaim represents cluster information that a managed cluster claims
+// ClusterClaims with well known names include,
+//   1. id.k8s.io, it contains a unique identifier for the cluster.
+//   2. clusterset.k8s.io, it contains an identifier that relates the cluster
+//      to the ClusterSet in which it belongs.
+type ClusterClaim struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Spec defines the attributes of the ClusterClaim.
+	Spec ClusterClaimSpec `json:"spec,omitempty"`
+}
+
+type ClusterClaimSpec struct {
+	// Value is a claim-dependent string
+	Value string `json:"value,omitempty"`
```

Comment:
you should have a length restriction at the very least.

Reply:
Length restriction added.